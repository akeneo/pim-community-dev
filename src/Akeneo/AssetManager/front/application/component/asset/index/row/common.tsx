import * as React from 'react';
import {NormalizedItemRecord} from 'akeneoreferenceentity/domain/model/record/record';
import {getImageShowUrl} from 'akeneoreferenceentity/tools/media-url-generator';
import {denormalizeFile} from 'akeneoreferenceentity/domain/model/file';
import {getLabel} from 'pimui/js/i18n';
import Completeness from 'akeneoreferenceentity/domain/model/record/completeness';
import CompletenessLabel from 'akeneoreferenceentity/application/component/app/completeness';

const memo = (React as any).memo;

const CommonRow = memo(
  ({
    record,
    locale,
    placeholder = false,
    onRedirectToRecord,
  }: {
    record: NormalizedItemRecord;
    locale: string;
    placeholder?: boolean;
  } & {
    onRedirectToRecord: (record: NormalizedItemRecord) => void;
  }) => {
    if (true === placeholder) {
      return (
        <tr>
          <td className="AknGrid-bodyCell AknGrid-bodyCell--image">
            <div className="AknGrid-bodyCellContainer AknLoadingPlaceHolder" />
          </td>
          <td className="AknGrid-bodyCell" colSpan={3}>
            <div className="AknGrid-bodyCellContainer AknLoadingPlaceHolder" />
          </td>
        </tr>
      );
    }

    const label = getLabel(record.labels, locale, record.code);

    return (
      <tr
        className="AknGrid-bodyRow"
        data-identifier={record.identifier}
        onClick={event => {
          event.preventDefault();

          onRedirectToRecord(record);

          return false;
        }}
      >
        <td className="AknGrid-bodyCell AknGrid-bodyCell--image">
          <img
            className="AknGrid-image AknLoadingPlaceHolder"
            width="44"
            height="44"
            src={getImageShowUrl(denormalizeFile(record.image), 'thumbnail_small')}
          />
        </td>
        <td className="AknGrid-bodyCell" title={label}>
          {label}
        </td>
        <td className="AknGrid-bodyCell AknGrid-bodyCell--identifier" title={record.code}>
          {record.code}
        </td>
        <td className="AknGrid-bodyCell">
          <CompletenessLabel completeness={Completeness.createFromNormalized(record.completeness)} expanded={false} />
        </td>
      </tr>
    );
  }
);

const CommonRows = memo(
  ({
    records,
    locale,
    placeholder,
    onRedirectToRecord,
    recordCount,
  }: {
    records: NormalizedItemRecord[];
    locale: string;
    placeholder: boolean;
    onRedirectToRecord: (record: NormalizedItemRecord) => void;
    nextItemToAddPosition: number;
    recordCount: number;
  }) => {
    if (placeholder) {
      const record = {
        identifier: '',
        reference_entity_identifier: '',
        code: '',
        labels: {},
        image: null,
        values: [],
        completeness: {},
      };

      const placeholderCount = recordCount < 30 ? recordCount : 30;

      return Array.from(Array(placeholderCount).keys()).map(key => (
        <CommonRow placeholder={placeholder} key={key} record={record} locale={locale} onRedirectToRecord={() => {}} />
      ));
    }

    return records.map((record: NormalizedItemRecord) => {
      return (
        <CommonRow
          placeholder={false}
          key={record.identifier}
          record={record}
          locale={locale}
          onRedirectToRecord={onRedirectToRecord}
        />
      );
    });
  }
);

export default CommonRows;
