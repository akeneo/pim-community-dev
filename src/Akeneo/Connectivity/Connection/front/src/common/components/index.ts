import * as EmptyState from '../../common/components/EmptyState';
import * as Typography from '../../common/components/Typography';
import {ApplyButton} from './button/ApplyButton';
import {GreyButton} from './button/GreyButton';
import {GhostButton} from './button/GhostButton';
import {IconButton} from './button/IconButton';
import {ImportantButton} from './button/ImportantButton';
import {DropdownLink, SecondaryActionsDropdownButton} from './button/SecondaryActionsDropdownButton';
import {Checkbox} from './Checkbox';
import {Figure, FigureCaption, FigureImage} from './Figure';
import {Form} from './form/Form';
import {FormGroup} from './form/FormGroup';
import {FormInput} from './form/FormInput';
import {Helper, HelperLink, HelperTitle} from './Helper';
import {Modal} from './Modal';
import {PageContent} from './PageContent';
import {PageError} from './PageError';
import {PageHeader} from './PageHeader';
import {RuntimeError} from './RuntimeError';
import {Section} from './Section';
import {Select} from './select/Select';
import {Select2, Select2Configuration} from './Select2';
import {SmallHelper} from './SmallHelper';
import {Table, TableCell, TableHeaderCell, TableHeaderRow, TableRow} from './Table';
import {ToggleButton} from './ToggleButton';
import {Flag} from './Flag';
import {Loading} from './Loading';

export {
    ApplyButton,
    Checkbox,
    DropdownLink,
    Figure,
    FigureCaption,
    FigureImage,
    Form,
    FormGroup,
    FormInput,
    GreyButton,
    GhostButton,
    Helper,
    HelperLink,
    HelperTitle,
    IconButton,
    ImportantButton,
    Modal,
    PageContent,
    PageError,
    PageHeader,
    RuntimeError,
    SecondaryActionsDropdownButton,
    Section,
    Select,
    Select2,
    Select2Configuration,
    SmallHelper,
    Table,
    TableCell,
    TableHeaderCell,
    TableHeaderRow,
    TableRow,
    ToggleButton,
    EmptyState,
    Typography,
    Flag,
    Loading,
};
