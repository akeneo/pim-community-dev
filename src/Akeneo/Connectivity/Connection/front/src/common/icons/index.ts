import {CaretDownIcon, CaretUpIcon} from './CaretIcon';

export {CaretDownIcon, CaretUpIcon};
