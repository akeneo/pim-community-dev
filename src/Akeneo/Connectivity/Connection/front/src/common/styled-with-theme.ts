import styled, {ThemedStyledInterface} from 'styled-components';
import searchIconUrl from './assets/icons/search.svg';
import graphIllustrationUrl from './assets/illustrations/Graph.svg';
import apiIllustrationUrl from './assets/illustrations/NewAPI.svg';
import surveyIllustrationUrl from './assets/illustrations/UserSurvey.svg';
import {pimTheme} from 'akeneo-design-system';

const theme = {
    ...pimTheme,
    color: {
        ...pimTheme.color,
        blue10: '#f5f9fc',
        blue100: '#5992c7',
        grey100: '#a1a9b7',
        grey120: '#67768a',
        grey140: '#11324d',
        grey20: '#f6f7fb',
        grey60: '#e8ebee',
        grey80: '#c7cbd4',
        purple100: '#9452ba',
        red100: '#d4604f',
        yellow10: '#fef7ec',
    },
    fontSize: {
        title: '30px',
        bigger: '17px',
        big: '15px',
        default: '13px',
        small: '11px',
        metricsBig: '22px',
    },
    icon: {
        search: searchIconUrl,
    },
    illustration: {
        api: apiIllustrationUrl,
        graph: graphIllustrationUrl,
        survey: surveyIllustrationUrl,
    },
};

type Theme = typeof theme;

export {theme, Theme};
export default styled as ThemedStyledInterface<Theme>;
