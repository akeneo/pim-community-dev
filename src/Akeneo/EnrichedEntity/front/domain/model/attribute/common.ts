import MinimalAttribute, {
  MinimalNormalizedAttribute,
  MinimalConcreteAttribute,
  AttributeType,
} from 'akeneoenrichedentity/domain/model/attribute/minimal';
import Identifier from 'akeneoenrichedentity/domain/model/attribute/identifier';
import AttributeCode from 'akeneoenrichedentity/domain/model/attribute/code';
import EnrichedEntityIdentifier from 'akeneoenrichedentity/domain/model/enriched-entity/identifier';
import LabelCollection from 'akeneoenrichedentity/domain/model/label-collection';

export interface CommonNormalizedAttribute extends MinimalNormalizedAttribute {
  order: number;
  required: boolean;
}

export interface CommonAttribute extends MinimalAttribute {
  order: number;
  required: boolean;
  normalize(): CommonNormalizedAttribute;
}

export abstract class CommonConcreteAttribute extends MinimalConcreteAttribute implements CommonAttribute {
  protected constructor(
    identifier: Identifier,
    enrichedEntityIdentifier: EnrichedEntityIdentifier,
    code: AttributeCode,
    labelCollection: LabelCollection,
    type: AttributeType,
    valuePerLocale: boolean,
    valuePerChannel: boolean,
    readonly order: number,
    readonly required: boolean
  ) {
    super(identifier, enrichedEntityIdentifier, code, labelCollection, type, valuePerLocale, valuePerChannel);

    if (typeof order !== 'number') {
      throw new InvalidArgumentError('Attribute expect a number as order');
    }
    if (typeof required !== 'boolean') {
      throw new InvalidArgumentError('Attribute expect a boolean as required value');
    }
  }

  public normalize(): CommonNormalizedAttribute {
    return {
      ...super.normalize(),
      order: this.order,
      required: this.required,
    };
  }
}

class InvalidArgumentError extends Error {}
