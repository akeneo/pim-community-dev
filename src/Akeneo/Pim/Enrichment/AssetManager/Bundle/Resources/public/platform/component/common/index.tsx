import styled from 'styled-components';
import {ThemedProps} from 'akeneoassetmanager/application/component/app/theme';

export const Pill = styled.div`
  background-color: ${(props: ThemedProps<void>) => props.theme.color.yellow100}
  width: 8px;
  min-width: 8px; // to fix a glitch on chrome when the pill is smashed
  height: 8px;
  border-radius: 8px;
  margin: 0 6px;
  align-self: center;
`;

export const Spacer = styled.div`
  flex: 1;
`;

export const Separator = styled.div`
  border-left: 1px solid ${(props: ThemedProps<void>) => props.theme.color.grey100};
  margin: 0 10px;
`;
