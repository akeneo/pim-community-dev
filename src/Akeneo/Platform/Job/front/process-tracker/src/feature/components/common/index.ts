export * from './JobExecutionStatus';
export * from './ProgressCell';
export * from './StopJobAction';
