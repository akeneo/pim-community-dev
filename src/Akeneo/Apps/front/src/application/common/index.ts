import {Breadcrumb} from './components/breadcrumb/Breadcrumb';
import {BreadcrumbItem} from './components/breadcrumb/BreadcrumbItem';
import {ApplyButton} from './components/button/ApplyButton';
import {Figure, FigureCaption, FigureImage} from './components/Figure';
import {Form} from './components/form/Form';
import {FormGroup} from './components/form/FormGroup';
import {FormInput} from './components/form/FormInput';
import {Helper, HelperLink, HelperTitle} from './components/Helper';
import {Loading} from './components/Loading';
import {Modal} from './components/Modal';
import {Page} from './components/Page';
import {PageHeader} from './components/PageHeader';
import {Section} from './components/Section';
import {Select2} from './components/Select2';

export {
    ApplyButton,
    Breadcrumb,
    BreadcrumbItem,
    Figure,
    FigureCaption,
    FigureImage,
    Form,
    FormGroup,
    FormInput,
    Helper,
    HelperLink,
    HelperTitle,
    Loading,
    Modal,
    Page,
    PageHeader,
    Section,
    Select2,
};
