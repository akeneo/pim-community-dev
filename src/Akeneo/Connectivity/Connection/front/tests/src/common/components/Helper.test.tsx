import React from 'react';
import {create} from 'react-test-renderer';
import {Helper, HelperLink, HelperTitle} from '@src/common';

describe('Helper', () => {
    it('should render', () => {
        const component = create(
            <Helper>
                <HelperTitle></HelperTitle>
                content
                <HelperLink />
            </Helper>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
