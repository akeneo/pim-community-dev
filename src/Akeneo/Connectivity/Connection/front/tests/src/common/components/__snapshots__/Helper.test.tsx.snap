// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper should render 1`] = `
<div
  className="AknDescriptionHeader"
>
  <div
    className="AknDescriptionHeader-icon"
    style={
      Object {
        "backgroundImage": "url('test-file-stub')",
      }
    }
  />
  <div
    className="AknDescriptionHeader-title"
  >
    <div
      className="AknDescriptionHeader-description"
    >
      content
      <a
        className="AknDescriptionHeader-link"
      />
    </div>
  </div>
</div>
`;
